app/services/idv/phone_step.rb

```diff
@@ -43,7 +43,11 @@ def async_state_done(async_state)
       @idv_result = async_state.result
 
       success = idv_result[:success]
-      handle_successful_proofing_attempt if success
+      if success
+        handle_successful_proofing_attempt
+      else
+        idv_session.add_failed_phone_step_number(idv_session.previous_phone_step_params[:phone])
```

Comment:
I don't think we should mark the number failed if the request to the vendor timed out (when `failure_reason == :timeout`), since the user could retry the same number and it might work. I think you can simulate this if you use `7035555888` as the phone number.

Reply:
thanks @matthinz  - i have added the distinction